src/main/java/org/folio/rest/utils/MigrationUtils.java

```diff
@@ -0,0 +1,164 @@
+package org.folio.rest.utils;
+
+import static org.folio.rest.impl.InvoiceStorageImpl.INVOICE_TABLE;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.ModuleId;
+import org.folio.okapi.common.SemVer;
+import org.folio.rest.core.RestClient;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
+import org.folio.rest.jaxrs.model.Invoice;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.CriterionBuilder;
+import org.folio.rest.persist.DBClient;
+
+public class MigrationUtils {
+  private static final Logger log = LogManager.getLogger(HelperUtils.class);
+
+  private MigrationUtils() {
+  }
+
+  public static Future<Void> migration(TenantAttributes attributes, String migrationModule, Supplier<Future<Void>> supplier) {
+    if (StringUtils.isBlank(attributes.getModuleFrom()))
+      return Future.succeededFuture();
+
+    SemVer moduleTo = moduleVersionToSemVer(migrationModule);
+    SemVer currentModuleVersion = moduleVersionToSemVer(attributes.getModuleFrom());
+    if (moduleTo.compareTo(currentModuleVersion) > 0) {
```

Comment:
logs to conditional statements should be added to investigate issues with migration if any

Reply:
Added logs, thanks